### PR TITLE
Now can also calculate wigner9j symbol for half integer input.

### LINKED ISSUE #13468: Problem with sympy.physics.wigner
When using the package wigner_9j from sympy.physics.wigner it gives wrong results for some specific wigner 9j symbols. The example given below does in fact only have integer or half-integer input and furthermore fulfills the triangle rule, yet it still returns an error. I cross checked with the module given on the webpage from [Anthony Stone]( http://www-stone.ch.cam.ac.uk/cgi-bin/wigner.cgi?symbol=9j&a=1%2F2&b=1%2F2&c=0&d=1%2F2&e=3%2F2&f=1&g=0&h=1&i=1 ) where you can see that the correct result should be 1/6. I also intrinsically found the same result when comparing with contact matrix elements for the scattering in hadronic molecules. Since the code is rather complicated I could not find the bug myself.

### Reproducing code example:
```
from sympy.physics.wigner import wigner_9j

wigner_9j(1/2,1/2,0, 1/2,3/2,1, 0,1,1 ,prec=16) 

```

### Error message:
```
raise ValueError("j values must be integer or half integer and fulfill the triangle relation")

ValueError: j values must be integer or half integer and fulfill the triangle relation
```

### PR DESCRIPTION
**Pull request by Trave11er**
The summation now runs over half-integers and a (-1)**(2*j_sum) factor
has been added for conversion between racah and wigner6j.
A doctest with an example input has been added. 

This fixes #13468.

### CODE REVIEW

File: sympy/physics/wigner.py [hunk 1]
```diff
@@ -533,16 +536,56 @@ def wigner_9j(j_1, j_2, j_3, j_4, j_5, j_6, j_7, j_8, j_9, prec=None):
...
     sumres = 0
-    for kk in range(imin, int(imax) + 1):
-        sumres = sumres + (2 * kk + 1) * \
-            racah(j_1, j_2, j_9, j_6, j_3, kk, prec) * \
-            racah(j_4, j_6, j_8, j_2, j_5, kk, prec) * \
-            racah(j_1, j_4, j_9, j_8, j_7, kk, prec)
+    for kk in range(0, 2*(int(imax) + 1)):
+        k = S.Half*kk
+        try:
+            j_sum = j_1+j_2+j_9+j_6+j_4+j_8
```
Comment (jksuom) [inline sympy/physics/wigner.py]: This could be taken out of the loop.

File: sympy/physics/wigner.py [hunk 2]
```diff
@@ -533,16 +536,56 @@ def wigner_9j(j_1, j_2, j_3, j_4, j_5, j_6, j_7, j_8, j_9, prec=None):
...
+            sumres +=(-1)**(kk+2*j_sum)*(kk + 1) * \
+            racah(j_1, j_2, j_9, j_6, j_3, k, prec) * \
+            racah(j_4, j_6, j_8, j_2, j_5, k, prec) * \
+            racah(j_1, j_4, j_9, j_8, j_7, k, prec)
+        except:
```
Comment (jksuom) [inline sympy/physics/wigner.py]: Bare excepts should not be used. It is only ValueError that is excepted and silently passed. If something else would turn up, that would be an error that should not be handled here.

**Comment by jksuom** [comment]
There is also trailing whitespace. You can find that by running `bin/test quality`.

File: sympy/physics/wigner.py (hunk 1 above)
Comment (Trave11er) [reply sympy/physics/wigner.py]: Sure, have done that.

File: sympy/physics/wigner.py (hunk 2 above)
Comment (Trave11er) [reply sympy/physics/wigner.py]: Thank you for explaining - changed to ValueError.

**Comment by jksuom** [comment]
There was a transient error in Travis. I've restarted the job.

**Comment by jksuom** [comment]
This looks good, thanks!

**Comment by Trave11er** [comment]
Thank you for your help :)

**Comment by certik** [comment]
There should be regular tests for this, not just doctests. That, together with a cleaner solution (I think) has been implemented in #13589.

**Pull request merged**